### ISSUE TITLE
Bug: `common_base` not found on Windows

### ISSUE THREAD

**Issue by hectormz** [econchick/interrogate#15]
First of all, I really like the intent of this package! 👍

**Is this a BUG REPORT or FEATURE REQUEST?**:

* [X] bug
* [ ] feature

### Environment

- `interrogate` version: 1.14
- Operating System(s): Windows 10
- Python version(s): 3.8

### What happened
`utils.get_common_base()` does not find the common base for files on Windows.

This happens because path levels are split by `\\` in Python on Windows. As a result, the `common_base` is an empty string (`""`). This has a downstream effect when generating the summary tables. 

Specifically, in `_get_filename()`, since the `common_base` is an empty string, a filepath:

`C:\path\to\project_root\src\file.py` 

becomes 
`:\path\to\project_root\src\file.py`

instead of:
 `\src\file.py`

### Suggestion
Since this is a python 3.5+ project, I think that this could be fixed in a flexible way by handling filepaths with [`pathlib`](https://docs.python.org/3/library/pathlib.html) instead. 
For example this in `utils.py`:
```python
level_slices = zip(*[f.split("/") for f in files])
```
could be replaced with:
```python
from pathlib import Path
level_slices = zip(*[Path(f).parts for f in files])
```
(if file names had not already been converted to `Path` objects)

Using `pathlib` here would help make this feature work across systems, but I think it could also replace just about all path manipulation `interrogate` uses `os` for.

If you'd like I could work contributing to any of these proposed changes.

**Comment by econchick** [comment]
Hey @hectormz ! Thanks so much for such a thorough bug report, it's very helpful. I'd welcome any contribution :) LMK if I can help at all.

**Comment by hectormz** [comment]
@econchick sure thing. I'll make a PR and see what you think!

Another thing that I found was that I had `attrs` version 19.1.0 installed, which causes an error when using `interrogate`. The error goes away if using `attrs>=19.2.0`. 

I can include that version requirement in `setup.py` during this PR unless you'd like a separate one.

**Comment by econchick** [comment]
Ah good catch - yea feel free to through that in the same PR (separate commit if possible though). I appreciate it!

**Comment by econchick** [comment]
So instead of fixing all the paths, I tried to keep the bug fix very narrow. Unfortunately I don't have an easy way to test on a windows machine. Would you mind checking out #18 for me? In the mean time, I'm going to try and get github actions to work w windows, but [it's not as easy as I thought...](https://github.com/econchick/interrogate/actions/runs/100091422)

**Comment by econchick** [comment]
@hectormz aha! I got windows testing to work on github! and found lots of test failures :D I am working through them :)

**Comment by hectormz** [comment]
> So instead of fixing all the paths, I tried to keep the bug fix very narrow. Unfortunately I don't have an easy way to test on a windows machine. Would you mind checking out #18 for me? In the mean time, I'm going to try and get github actions to work w windows, but [it's not as easy as I thought...](https://github.com/econchick/interrogate/actions/runs/100091422)

Yes, let me get to my computer and test this out for you.

**Comment by hectormz** [comment]
> @hectormz aha! I got windows testing to work on github! and found lots of test failures :D I am working through them :) 

@econchick awesome! When I was working through this I found it easier to test on a Linux VM then figure out the Windows test failures 😁. I assumed they might be due to the posix-style path patterns in the tests?

**Comment by econchick** [comment]
Addressed by #18 (thanks again @hectormz!)
